protocol/app/flags/flags.go

```diff
@@ -85,24 +87,34 @@ func GetFlagValuesFromOptions(
 	}
 
 	// Populate the flags if they exist.
-	if v, ok := appOpts.Get(NonValidatingFullNodeFlag).(bool); ok {
-		result.NonValidatingFullNode = v
+	if option := appOpts.Get(NonValidatingFullNodeFlag); option != nil {
+		if v, err := cast.ToBoolE(option); err == nil {
+			result.NonValidatingFullNode = v
+		}
```

Comment:
we can maybe just do `result.NonValidatingFullNode = cast.ToBool(option)` for simplicity (i.e. ignore error) but let's worry about that later

Reply:
I wonder if it would be more useful to throw an error or panic if the flag value can't be converted.

If I was an operator and I accidentally typed `tru` insted of `true`, I would want to software to fail and tell me my input was incorrect rather than using the default `false` value.

wdyt?

Reply:
yup Brendan brought this up too - we decided to punt this to M6 (since our current behavior can't handle this either)

ticket is [here](https://linear.app/dydx/issue/DEC-2146)